sunpy/instr/fermi.py

```diff
@@ -0,0 +1,373 @@
+    contains 1 minute cadence data on the spacecraft pointing, useful for calculating detector angles.
+
+    Parameters
+    ----------
+
+    date : datetime object
+        A datetime object or other date format understood by the parse_time function.
+    '''
+
+    date = parse_time(date)
+    #use a temp directory to hold the file
+    tmp_dir=tempfile.mkdtemp()
+    #use Fermi data server to access weekly LAT pointing file.
+    base_url = 'http://fermi.gsfc.nasa.gov/ssc/observations/timeline/ft2/files/'
+    fbasename='FERMI_POINTING_FINAL_'
+    
+    #find out which file to get based on date
+    #earliest file in the FERMI server is for mission week 23, beginning 2008 November 6.
+    weekly_file_start=parse_time('2008-11-06')
+    base_week=23
+
+    #find out which mission week corresponds to date
+    time_diff=date-weekly_file_start
+    weekdiff = time_diff.days//7
+    week = weekdiff + base_week
+
+    #find out the rest of the file name. Need the year and the day-in-year for start and end of file
+    start_date = weekly_file_start + datetime.timedelta(weekdiff*7)
+    start_year_str=str(start_date.year) + '-01-01'
+    start_str = start_date.strftime('%Y%j')
+
+    #now end string
+    end_date = weekly_file_start + datetime.timedelta((weekdiff+1)*7)
+    end_str = end_date.strftime('%Y%j')
+
+    #construct the full url for the weekly pointing file
+    full_fname_start=fbasename + str(week) + '_' + start_str + '_' + end_str + '_'
+    full_fname_extension='.fits'
+    #the full filename will be full_fname_start + version number + full_fname_extension, but version number unknown
+    #multiple versions may exist for each week.
+
+    #Parse the base_url page for all file links. Find all matching files for the desired week
+    resp=urllib2.urlopen(base_url)
+    #get the returned html as a string
+    html_string=resp.read().decode('utf-8')
+    #parse the html string using BeautifulSoup 
+    parsed_html = BeautifulSoup(html_string,'html5lib')
+    #find all the links in the html
+    links=parsed_html.body.findAll('a')
+
+    #find all files matching the desired week
+    matching_files = [l.text for l in links if (l.text.startswith(full_fname_start) and l.text.endswith(full_fname_extension))]
+    if not matching_files:
+        raise ValueError('No Fermi pointing files found for given date!')
+    #find the file with the highest version number
+    matching_files.sort()
+    #this is the correct pointing file
+    full_fname=matching_files[-1]
+    
+    #download the file
+    pointing_file_url=urlparse.urljoin(base_url,full_fname)
+    destination=os.path.join(tmp_dir,full_fname)
+    urllib.urlretrieve(pointing_file_url,destination)
+
+    #return the location of the downloaded file
+    return destination
+
+
+def get_detector_sun_angles_for_time(time, file):
+    '''get the GBM detector angles vs the sun for a single time.
```

Comment:
(minor) It is conventional to use tripe double quotes, and to put the text on the line after the open quotes:

```python
"""
get the ...
"""